modules/lightning_features/lightning_api/lightning_api.module

```diff
@@ -2,35 +2,27 @@
 
 use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Url;
-use Drupal\lightning_api\OAuthHelper;
 use Symfony\Component\Routing\Exception\RouteNotFoundException;
 
 /**
  * Implements hook_modules_installed().
  */
 function lightning_api_modules_installed(array $modules) {
-  // Don't do anything during a config sync.
-  if (\Drupal::isConfigSyncing()) {
-    return;
-  }
-  elseif (in_array('simple_oauth', $modules)) {
+  if (in_array('simple_oauth', $modules)) {
+    $url = Url::fromRoute('lightning_api.generate_keys');
     try {
-      list ($private, $public) = OAuthHelper::generateKeyPair(TRUE);
-
-      \Drupal::configFactory()
-        ->getEditable('simple_oauth.settings')
-        ->set('private_key', $private)
-        ->set('public_key', $public)
-        ->save();
-
-      drupal_set_message(t('Generated an initial key pair for OAuth authentication. To keep your site secure, be sure to change it before going into production!'), 'warning');
+      $url = $url->toString();
     }
-    catch (\Exception $e) {
-      $error = t('Could not generate a key pair because @reason. Until this is done, OAuth authentication will not work.', [
-        '@reason' => lcfirst($e->getMessage()),
-      ]);
-      drupal_set_message($error, 'error');
+    catch (RouteNotFoundException $e) {
+      \Drupal::service('router.builder')->rebuild();
     }
+
+    drupal_set_message(
+      t('You may wish to <a href=":generate_keys">generate a key pair</a> for OAuth authentication.', [
```

Comment:
If installing through the UI, this message is displayed on the first page load and it ends up linking to "/core/install.php/admin/config/system/lightning/api/keys" as a result (note the install script in the path).